src/s3/explorer/s3Nodes.ts

```diff
@@ -38,6 +39,31 @@ export class S3Node extends AWSTreeNodeBase {
         })
     }
 
+    private async listBucketNodes(): Promise<(S3BucketNode | AWSCommandTreeNode)[]> {
+        try {
+            const response = await this.s3.listBuckets()
+            return response.buckets.map(bucket => new S3BucketNode(bucket, this, this.s3))
+        } catch (err) {
+            if (err.code === 'AccessDenied') {
+                if (this.addedBuckets.length === 0) {
+                    const localizedText = localize('AWS.explorerNode.s3.showBucket', 'Click to show existing bucket')
+                    return [new AWSCommandTreeNode(this, localizedText, 'aws.s3.showBucket', [this])]
+                } 
+
+                return this.addedBuckets
+            } else {
+                throw err
+            }
+        }
+    }
+
+    public async showBucket(bucketName: string): Promise<void> {
+        await this.s3.listFiles({ bucketName })
+        // For now, there isn't a good way to get a bucket's region + ARN from its name.
+        const bucket = { name: bucketName, region: this.regionCode, arn: `arn:aws:s3:::${bucketName}` }
```

Comment:
So my understanding is, this will throw the bucket in any region that you click as long as you have access? I feel like this warrants a global node.

Reply:
No, `listFiles` throws an error if the bucket isn't in the same region as the client.